common4j/src/main/com/microsoft/identity/common/java/commands/parameters/nativeauth/AcquireTokenNoFixedScopesCommandParameters.java

```diff
@@ -0,0 +1,156 @@
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.java.commands.parameters.nativeauth;
+
+import com.google.gson.annotations.Expose;
+import com.microsoft.identity.common.java.authorities.CIAMAuthority;
+import com.microsoft.identity.common.java.authscheme.AbstractAuthenticationScheme;
+import com.microsoft.identity.common.java.dto.IAccountRecord;
+import com.microsoft.identity.common.java.exception.ArgumentException;
+import com.microsoft.identity.common.java.exception.ClientException;
+import com.microsoft.identity.common.java.exception.TerminalException;
+import com.microsoft.identity.common.java.logging.Logger;
+import com.microsoft.identity.common.java.providers.microsoft.azureactivedirectory.AzureActiveDirectory;
+import com.microsoft.identity.common.java.providers.microsoft.azureactivedirectory.AzureActiveDirectoryCloud;
+
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.experimental.SuperBuilder;
+
+@Getter
+@EqualsAndHashCode(callSuper = true)
+@SuperBuilder(toBuilder = true)
+public class AcquireTokenNoFixedScopesCommandParameters extends BaseNativeAuthCommandParameters {
+
+    private static final String TAG = AcquireTokenNoFixedScopesCommandParameters.class.getSimpleName();
+
+    private static final Object sLock = new Object();
+
+    private final IAccountRecord account;
+
+    @Expose()
+    private final Set<String> scopes;
+
+    @Expose()
+    private final String claimsRequestJson;
+
+    @Expose()
+    private final AbstractAuthenticationScheme authenticationScheme;
+
+    @Expose()
+    private final String mamEnrollmentId;
+
+    @Expose()
+    private final boolean forceRefresh;
+
+    private final String loginHint;
+
+    private final List<Map.Entry<String, String>> extraOptions;
+
+    public Set<String> getScopes() {
+        return this.scopes == null ? null : new HashSet<>(this.scopes);
+    }
+
+    public String getMamEnrollmentId(){
```

Comment:
If you are adding this class anyway, then you should ask yourself: Do you imagine using MAM Enrollment Id in Native Auth flow? Why or why not? If you don't need it then you should you still have mamEnrollmentId as a field in this class?

Reply:
Good point, this can be improved. Looking at how we use this field: we're passing it to the cache when retrieving data from the cache through this method: 
![image](https://github.com/AzureAD/microsoft-authentication-library-common-for-android/assets/1315628/bddb702f-8e67-4490-bdde-72f9fdb5e486)

am I right to conclude that we can omit the `mamEnrollmentIdentifier` parameter? If so, we can clean it up from this CommandParameters class 👍

Reply:
It depends on whether you want to support MAM feature or not? Probably not since MAM today requires Broker and Device Registration.

Reply:
This method has been removed.